TheWarWithin/RogueAssassination.lua

```diff
@@ -2434,27 +2432,36 @@ spec:RegisterAbilities( {
                 debuff.rupture_deathmark.pmultiplier = persistent_multiplier
             end
 
-            if buff.indiscriminate_carnage_any.up then
-                active_dot.rupture = min( true_active_enemies, active_dot.rupture + 2 )
-            end
-
-            if buff.finality_rupture.up then removeBuff( "finality_rupture" )
-            elseif talent.finality.enabled then applyBuff( "finality_rupture" ) end
-
+            spend( combo_points.current, "combo_points" )
+            if talent.supercharger.enabled then removeStack( "supercharged_combo_points" ) end
 
+            --- Assassination Rogue specific
 
             if talent.scent_of_blood.enabled or azerite.scent_of_blood.enabled then
                 applyBuff( "scent_of_blood", dot.rupture.remains, active_dot.rupture )
             end
 
-            spend( combo_points.current, "combo_points" )
-            removeStack( "supercharged_combo_points" )
+            if buff.indiscriminate_carnage_any.up then
+                active_dot.rupture = min( true_active_enemies, active_dot.rupture + 2 )
+            end
 
             if buff.serrated_bone_spike_charges.up then
-                gain ( 1 + buff.serrated_bone_spike_charges.stack, "combo_points" )
-                removeStack( "serrated_bone_spike_charges" )
+                for i = 1, buff.indiscriminate_carnage_any.up and 3 or 1 do
+                    gain ( 1 + buff.serrated_bone_spike_charges.stack, "combo_points" )
+                    removeStack( "serrated_bone_spike_charges" )
+                end
                 applyDebuff( "target", "serrated_bone_spike_dot" )
+                if buff.indiscriminate_carnage_any.up then active_dot.serrated_bone_spike_dot = min ( true_active_enemies, active_dot.serrated_bone_spike_dot + 2 ) end
             end
+
+            --- Subtlety Rogue specific
+
+            if spec.id == 261 then
```

Comment:
`state.spec.subtlety` instead of `spec.id == 261` -- there's a subtle issue that `spec` in this file is the spec (Assassination) model itself (i.e., `spec:RegisterAbilities`), rather than the `spec` table in `state` (`state.spec`).